src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -674,7 +674,7 @@ def _auto_select_batch_size(self) -> int:
             try:
                 train_dataloader = train_dl_source.dataloader()
                 if hasattr(train_dataloader, "batch_sampler"):
-                    batch_size = train_dataloader.batch_sampler.batch_size  # type: ignore[union-attr]
+                    batch_size = train_dataloader.batch_sampler.batch_size  # type: ignore[union-attr, assignment]
```

Comment:
Once I added Instance as a return type in `data_connector._DataLoaderSource.dataloader`, this error was raised by mypy.

```sh
src/pytorch_lightning/strategies/deepspeed.py:677: error: Incompatible types in assignment (expression has type "Union[int, Any, Tensor, Module]", variable has type "int")  [assignment]
```

I traced back to `data_connector._DataLoaderSource.dataloader`, and there is no batch_sampler or batch_size in that specific method, or in the entire module.

`batch_sampler` and `batch_size` are attributes of torch.utils.data.DataLoader; given this is outside of the scope of PL, I opted to ignore the error.